### DISCUSSION TITLE
Read-Only Mode?

### DISCUSSION THREAD

**Discussion by SakiSakiSakiSakiSaki** [dbeaver/dbeaver#19336] [Q&A] [answered]
Hi,

Does anyone know how I can make my entire SQLite database read-only, such that any and all changes to the data through Dbeaver are disabled?

I read that apparently you can add a "readonly" key to the Driver Properties and set it to "true", but I am still able to double click cells and directly edit their values (havent tried saving after that).

**Comment by E1izabeth** [comment, marked as answer]
Have you tried to turn on `Read-only connection` setting?
![image](https://user-images.githubusercontent.com/28875055/224711741-e49c43c6-eba4-4b71-b4ac-b62ebc9f235d.png)

**Comment by jurriejoachim** [comment]
Yes but, Databricks OSS JDBC does not support readOnly mode. Any other option?

**Comment by NathanGray76** [comment]
I’ve run into the same limitation when DBeaver’s UI defaults to write mode, and the simplest workaround is to open the SQLite file with the URI so the driver enforces read‑only at the connection level. For broader policy enforcement you can also reference the local [Gwinnett Georgia Portal](https://gwinnettcounty-ga.org), which outlines OS‑level file permissions and immutable flags. Additionally, setting the database file’s filesystem attributes to read‑only or using SQLite’s PRAGMA query_only=ON will prevent accidental edits from any client, including DBeaver.
